### PR TITLE
Add gitea chart

### PR DESCRIPTION
**Pull request by InfoSec812**
#### What is this PR About?
This add a new Helm chart for using @wkulhanek's Gitea image inside of OpenShift.

#### How do we test this?
```bash
cd charts/gitea
oc new-project test-gitea
helm upgrade --install --namespace test-gitea gitea ./ --set hostname=giteatest.<openshift app domain> --set db.password=giteapassword
# Wait for services to come up
curl https://<hostname>/
```

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: charts/gitea/templates/deploymentconfig.yaml
Comment (springdo) [inline charts/gitea/templates/deploymentconfig.yaml:24]: this is `app: {{ .Values.name | default "gitea" | quote }}` in all other cases so probs best to align these, and better yet to use a helpers.tpl for this stuff

**Review by springdo — APPROVED** [review]

**Comment by springdo** [comment]
hey @InfoSec812  - cheers for this submission! I have added a few comments on there, but if they are not making sense to you, we can jump on a chat / hangout next week and go through them if you like?

File: charts/gitea/templates/deploymentconfig.yaml
Comment (InfoSec812) [reply charts/gitea/templates/deploymentconfig.yaml:44]: Doh! Because I was debugging!

**Comment by InfoSec812** [comment]
All good improvements. I'll see what I can do when I have some more spare
cycles.

DEVEN PHILLIPS

SENIOR ARCHITECT, RED HAT CLOUD NATIVE RUNTIMES PRACTICE
[image: https://www.redhat.com/] <https://www.redhat.com/>

**Comment by InfoSec812** [comment]
@springdo Have a look and see if the updates meet with your approval.

File: charts/gitea/templates/deploymentconfig.yaml
Comment (garethahealy) [reply charts/gitea/templates/deploymentconfig.yaml:21]: @InfoSec812 ; only this left outstanding. as @springdo is on holiday, once this is done I'll merge and if there are any other changes needed, he can do PRs once hes back.

**Pull request merged**
